python/cudf/cudf/core/_internals/timezones.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+import os
+import zoneinfo
+from functools import lru_cache
+
+from cudf._lib.timezone import build_timezone_transition_table
+from cudf.core.dataframe import DataFrame
+
+
+@lru_cache(maxsize=20)
```

Comment:
Is there a reason this should be 20? I'd prefer a larger value (default for `lru_cache` is 128). Do the objects in the cache consume GPU memory? How large do they tend to be?

Reply:
We can increase this to 128 (or even `None`). The objects in the cache do consume GPU memory, but even if we read _all_ the timezone data and cached the results, that would only take ~7 MB:

```
   ...: stats_mr = rmm.mr.StatisticsResourceAdaptor(rmm.mr.get_current_device_resource())
   ...: rmm.mr.set_current_device_resource(stats_mr)
   ...: print(stats_mr.allocation_counts)
   ...: dfs = []
   ...: for root, dirs, files in os.walk("/usr/share/zoneinfo"):
   ...:     for f in files:
   ...:         zone_name = ("/".join([root, f])).lstrip("/usr/share/zoneinfo")
   ...:         try:
   ...:             dfs.append(get_tz_data(zone_name))
   ...:         except Exception:  # not a zone name
   ...:             pass
   ...: print({key: format_bytes(val) for key, val in stats_mr.allocation_counts.items()})

{'current_bytes': 0, 'current_count': 0, 'peak_bytes': 0, 'peak_count': 0, 'total_bytes': 0, 'total_count': 0}
{'current_bytes': '7.48 MiB', 'current_count': '1.13 kiB', 'peak_bytes': '7.48 MiB', 'peak_count': '1.13 kiB', 'total_bytes': '7.48 MiB', 'total_count': '1.13 kiB'}
```

The total _number_ of time zones (at least on my machine) was 578. Should we increase the cache to something around that number, or `None`?

Reply:
Yeah, let’s go for None, or 1024 if you want an upper bound for some kind of safety?

Reply:
Sure